src/createServer.js

```diff
@@ -1,3 +1,49 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase/index');
+const { setError } = require('./setError');
+
+const createServer = () => {
+  const server = http.createServer((request, response) => {
+    const normalizedURL = new URL(request.url, `http://${request.headers.host}`);
+
+    const { pathname, searchParams } = normalizedURL;
+
+    const originalText = pathname.slice(1);
+    const targetCase = searchParams.get('toCase');
+
+    const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+    const isSupportedCase = targetCase
+      ? supportedCases.includes(targetCase.toUpperCase())
+      : false;
+
+    if (!originalText || !targetCase || !isSupportedCase) {
+      const responseError = setError(originalText, targetCase, isSupportedCase);
+
+      response.statusCode = 400;
+      response.setHeader('Content-Type', 'application/json');
+      response.end(JSON.stringify(responseError));
+
+      return;
```

Comment:
You can use the errors array to check if the errors exist.

```suggestion
      const errors = setError(originalText, targetCase);
      
      if (errors.length) {
        response.statusCode = 400;
        response.end(JSON.stringify(responseError));

        return;
      }
```